src/main/kotlin/frc/robot/subsystems/intake/roller/RollerIOSim.kt

```diff
@@ -19,8 +35,35 @@ class RollerIOSim : RollerIO {
         )
     private val controlRequest = VoltageOut(0.0)
 
+    private fun outtakeGamePieceIfNeed() {
+        if (intakeSimulation.obtainGamePieceFromIntake()) {
+            SimulatedArena.getInstance()
+                .addGamePieceProjectile(
+                    ReefscapeAlgaeOnFly(
+                        driveSimulation!!.simulatedDriveTrainPose.translation,
+                        Translation2d(
+                            Units.Meters.of(0.3),
+                            Units.Meters.zero()
+                        ),
+                        driveSimulation
+                            .driveTrainSimulatedChassisSpeedsFieldRelative,
+                        driveSimulation.simulatedDriveTrainPose.rotation,
+                        Units.Meter.of(0.4),
+                        Units.MetersPerSecond.of(3.0),
+                        Units.Degrees.of(0.0)
+                    )
+                )
+        }
+    }
+
     override fun setVoltage(voltage: Voltage) {
         motor.setControl(controlRequest.withOutput(voltage))
+        if (voltage != Units.Volts.zero()) {
```

Comment:
```suggestion
        if (voltage > Units.Volts.zero()) {
```
Feels more natural to me.

Reply:
Voltage can be negative.